scriptmodules/libretrocores/lr-flycast.sh

```diff
@@ -69,20 +69,22 @@ function install_lr-flycast() {
 }
 
 function configure_lr-flycast() {
-    mkRomDir "dreamcast"
-    ensureSystemretroconfig "dreamcast"
-
+    local sys
+    for sys in dreamcast arcade; do
+        mkRomDir "$sys"
+        ensureSystemretroconfig "$sys"
+        addSystem "$sys"
+        # platform-specific
+        if isPlatform "gl"; then
+            iniConfig " = " "" "$configdir/$sys/retroarch.cfg"
+            iniSet "video_shared_context" "true"
+        fi
+    done
     mkUserDir "$biosdir/dc"
 
-    # system-specific
-    if isPlatform "gl"; then
-        iniConfig " = " "" "$configdir/dreamcast/retroarch.cfg"
-        iniSet "video_shared_context" "true"
-    fi
-
     local def=0
     isPlatform "kms" && def=1
     # segfaults on the rpi without redirecting stdin from </dev/null
     addEmulator $def "$md_id" "dreamcast" "$md_inst/flycast_libretro.so </dev/null"
-    addSystem "dreamcast"
+    addEmulator 0 "$md_id" "arcade" "$md_inst/flycast_libretro.so </dev/null"
```

Comment:
addSystem should be called after addEmulator.

Note I am changing the setting of `video_shared_context` in this file as part of another PR (#3404) which I intend to look at again and update, so it may just be easier for me to implement this later or we can revisit when the other PR is done.

Reply:
I see, because in "remove" mode, addSystem will remove it from the FE (ex: `es_systems.cfg`), but only if addEmulator has already removed the system's `emulators.cfg` first, yeah?

Reply:
I see, because in "remove" mode, addSystem will remove it from the FE (ex: `es_systems.cfg`), but only if addEmulator has already removed the system's `emulators.cfg` first, yeah?

Reply:
Yep. That's correct.

Reply:
Based on your WIP. What do you think of:

```
function configure_lr-flycast() {
    local sys
    local systems=(dreamcast arcade)
    for sys in $systems; do
        local def=0
        if isPlatform "kms" && [[ "$sys" == "dreamcast" ]]; then
            def=1
        fi
        # segfaults on the rpi without redirecting stdin from </dev/null
        addEmulator $def "$md_id" "$sys" "$md_inst/flycast_libretro.so </dev/null"
        addSystem "$sys"
    done

    [[ "$md_mode" == "remove" ]] && return

    for sys in $systems; do
        mkRomDir "$sys"
        ensureSystemretroconfig "$sys"

    # platform-specific
    if isPlatform "gl"; then
        iniConfig " = " "" "$configdir/$sys/retroarch.cfg"
        iniSet "video_shared_context" "true"
    fi
    done

    mkUserDir "$biosdir/dc"
}
```

Or, with `defaultRASystemConfig`:

```
function configure_lr-flycast() {
    local sys
    local systems=(dreamcast arcade)
    local def
    for sys in $systems; do
        def=0
        if isPlatform "kms" && [[ "$sys" == "dreamcast" ]]; then
            def=1
        fi
        # segfaults on the rpi without redirecting stdin from </dev/null
        addEmulator $def "$md_id" "$sys" "$md_inst/flycast_libretro.so </dev/null"
        addSystem "$sys"
    done

    [[ "$md_mode" == "remove" ]] && return

    local params=()
    # system-specific
    if isPlatform "gl"; then
        params+=("video_shared_context" "true")
    fi
    for sys in $systems; do
        mkRomDir "$sys"
        defaultRASystemConfig "$sys" "${params[@]}"
    done

    mkUserDir "$biosdir/dc"
}
```

Is it okay to have "local" inside the "for" loop like in the first one, or should I put `local def` outside the loop and only have `def=0` inside it like in the second?

Do I need quotes on `systems=("dreamcast" "arcade")` or, (probably not on `sys in "$systems"`, that would pass it as a single arg, right?)